### PR TITLE
Js apitests

### PR DESCRIPTION
**Pull request by jacobstringer**
API tests added in. Changes to app.py include turning off some functions in the create game if testing, and updating comments.

### CODE REVIEW

File: application/app.py [hunk 1]
```diff
@@ -1,297 +1,294 @@
...
-    game = get_game(player=player)
-    game.complete = True
-    db.session.add(game)
-
-    try:
-        print('Ending game')
-        db.session.commit()
-        return jsonify(message="Success")
-
-    except Exception as e:
```
Comment (mmdonaldson) [inline application/app.py:292]: This looks like it's deleting the catch block

File: tests/test_api.py [hunk 2]
```diff
@@ -86,3 +88,300 @@ def test_protected_route(self):
...
+
+    def setUp(self):
+        db.create_all()
+
+    def tearDown(self):
+        db.session.remove()
+        db.drop_all()
+
+    def test_check_game(self):
+        user = User(email='test@test.com', name='test', password='tester')
```
Comment (mmdonaldson) [inline tests/test_api.py:107]: We probably need the email to generate randomly

Comment (mmdonaldson) [reply tests/test_api.py:107]: Or are these just getting a user that's already created? Do we have a function which makes sure they are created?

File: application/app.py (hunk 1 above)
Comment (jacobstringer) [reply application/app.py:292]: Good spotting

File: tests/test_api.py (hunk 2 above)
Comment (jacobstringer) [reply tests/test_api.py:107]: There needs to be a user and game already created to be able to create a player object. It doesn't really matter what they are beyond that.

Comment (mmdonaldson) [reply tests/test_api.py:107]: We will need to create the user/game objects because when this runs in memory in CI they won't be any previous data

**Review by mmdonaldson — APPROVED** [review]

**Pull request merged**